### DISCUSSION TITLE
is it possible to create a detecor to check how a constant is used in a code ?

### DISCUSSION THREAD

**Discussion by malawadd** [Cyfrin/aderyn#380] [Ideas]
this issue for example 

https://solodit.xyz/issues/chainlink-oracle-returns-stale-price-due-to-chainlink_timeout-variable-in-libchainlinkoracle-being-set-to-4-hours-codehawks-beanstalk-part-1-git

**Comment by TilakMaddy** [comment]
Take a look at the AST of the contract with constants. Then find out what that looks like in the `WorkspaceContext`. : )

<details><summary>Hint</summary>VariableDeclaration</details>

**Reply by malawadd** [reply]
I see , so is it recommended to build those sort of detectors that only work on some very specific and niche issue ? I imagine the list might be super long haha.

**Comment by TilakMaddy** [comment]
Closing due to staleness
